net/node/link.go

```diff
@@ -37,36 +37,43 @@ type link struct {
 func unpackNodeBuf(node *node, buf []byte) {
 	var msgLen int
 	var msgBuf []byte
-	if node.rxBuf.p == nil {
-		if len(buf) < MSGHDRLEN {
-			log.Warn("Unexpected size of received message")
-			errors.New("Unexpected size of received message")
-			return
-		}
 
-		length := 0
-		length, buf = PayloadLen(buf)
+	if len(buf) == 0 {
+		return
+	}
 
-		if length == 0 && buf == nil {
+	if node.rxBuf.len == 0 {
+		length := MSGHDRLEN - len(node.rxBuf.p)
+		if length > len(buf) {
+			length = len(buf)
+			node.rxBuf.p = append(node.rxBuf.p, buf[0:length]...)
 			return
 		}
 
-		msgLen = length + MSGHDRLEN
-	} else {
-		msgLen = node.rxBuf.len
+		node.rxBuf.p = append(node.rxBuf.p, buf[0:length]...)
+		if msg.ValidMsgHdr(node.rxBuf.p) == false {
+			node.rxBuf.p = nil
+			node.rxBuf.len = 0
+			log.Warn("Get error message header")
+			// TODO Relocate the message header
```

Comment:
It needs the relocate message header method here to update the relative info as the TODO said, 

It do go through the HandleNodeMsg and unpackNodeBuf, but the case will finally be catch as the buffer length 0, so it won't corrupt other data and introduce more side effect. 

But I agree to add a return here to explicit stopping the processing before a better message header relocate method available